### ISSUE TITLE
attempting to add contributor who is already a contributor throws 500 error

### ISSUE THREAD

**Issue by pdpinch** [mitodl/open-discussions#1634]
DoesNotExist: ChannelGroupRole matching query does not exist.

https://sentry.io/mit-office-of-digital-learning/discussions/issues/830607468/?referrer=github_plugin

```
DoesNotExist: ChannelGroupRole matching query does not exist.
(17 additional frame(s) were not displayed)
...
  File "rest_framework/serializers.py", line 214, in save
    self.instance = self.create(validated_data)
  File "channels/serializers.py", line 822, in create
    return api.add_contributor(username, channel_name)
  File "channels/api.py", line 1232, in add_contributor
    add_user_role(proxied_channel.channel, ROLE_CONTRIBUTORS, user)
  File "channels/api.py", line 357, in add_user_role
    get_role_model(channel, role).group.user_set.add(user)
  File "channels/api.py", line 345, in get_role_model
    return ChannelGroupRole.objects.get(channel=channel, role=role)

DoesNotExist: ChannelGroupRole matching query does not exist.
```

**Comment by pdpinch** [comment]
This is not the cause of this error. 

@mbertrand can you take a look at this auth model to see if there's a data inconsistency?

**Comment by mbertrand** [comment]
The `channels.api.get_role_model` function used to call `ChannelGroupRole.objects.get_or_create` but looks like it was changed to just `ChannelGroupRole.objects.get` about a month ago.  

The `backpopulate_subscriptions_roles` command does not create a `ChannelGroupRole` object for a particular channel/role if there are no users with that role.  In this case, the channel `3086xalums` probably did not have any contributors when the script was run.  Running it again won't work because of the above change.

**Comment by mbertrand** [comment]
@rhysyngsun do you know what the reason was for the change from `get_or_create` to `get`?  Rather than revert that, I was thinking of leaving it as is and modifying the `backpopulate_subscriptions_roles`command instead to create missing `ChannelGroupRole` objects there and populate them with users if any.  New channels should already create `ChannelGroupRole` objects for both contributors and moderators.

**Comment by rhysyngsun** [comment]
@mbertrand those records now get created atomically [here](https://github.com/mitodl/open-discussions/blob/master/channels/api.py#L603) so this state isn't possible for new channels. I had been under the assumption that we had previously backpopulated those records, but sounds like that may not have been the case.

**Comment by mbertrand** [comment]
There are two channels missing the `ChannelGroupRole` object for contributors (`3086xalums`, `react2019`) What do you think of just creating them manually in the django shell?

**Comment by mbertrand** [comment]
I manually added and populated the `ChannelGroupRole` objects for the two channels.  However, if we ever need to run the `backpopulate_subscriptions_roles` command again it will need to be modified before it will work.

**Comment by pdpinch** [comment]
@mbertrand was this closed by #1679 ?

**Comment by mbertrand** [comment]
Yes
